src/context/mod.rs

```diff
@@ -27,29 +34,71 @@ use crate::{
     AggregatedQueries, Config,
 };
 
-#[derive(Copy, Clone)]
-pub struct FileRunContext<'a> {
+pub struct FileRunContext<
+    'a,
+    'b,
+    TFromFileRunContextInstanceProviderFactory: FromFileRunContextInstanceProviderFactory,
+> {
     pub path: &'a Path,
     pub file_contents: RopeOrSlice<'a>,
     pub tree: &'a Tree,
-    pub config: &'a Config,
+    pub config: &'a Config<TFromFileRunContextInstanceProviderFactory>,
     pub language: SupportedLanguage,
-    pub(crate) aggregated_queries: &'a AggregatedQueries<'a>,
+    pub(crate) aggregated_queries:
+        &'a AggregatedQueries<'a, TFromFileRunContextInstanceProviderFactory>,
     pub(crate) query: &'a Arc<Query>,
-    pub(crate) instantiated_rules: &'a [InstantiatedRule],
+    pub(crate) instantiated_rules:
+        &'a [InstantiatedRule<TFromFileRunContextInstanceProviderFactory>],
+    from_file_run_context_instance_provider:
+        &'b TFromFileRunContextInstanceProviderFactory::Provider<'a>,
+}
+
+impl<
+        'a,
+        'b,
+        TFromFileRunContextInstanceProviderFactory: FromFileRunContextInstanceProviderFactory,
+    > Copy for FileRunContext<'a, 'b, TFromFileRunContextInstanceProviderFactory>
```

Comment:
At one point at least `#[derive(Copy)]` seemed to be doing some weird complaining about lifetimes that didn't make sense so I switched to a manual implementation